### DISCUSSION TITLE
Feature request: add the same SearchBox options to eg RefinementList SearchBox

### DISCUSSION THREAD

**Discussion by WHITE-developer** [algolia/instantsearch#5809] [Ideas]
The react [SearchBox widget](https://www.algolia.com/doc/api-reference/widgets/search-box/react/) has a lot of customization options, like setting the submitIconComponent etc.
These seems to be missing in the [RefinementList](https://www.algolia.com/doc/api-reference/widgets/refinement-list/react/).

Can we get the same options from the SearchBox to the RefinementList? 
Or is there some other way to achieve this? 
It looks like the [same component ](https://github.com/algolia/instantsearch/blob/c4f9eeb49dffb9caa12acef7ffa895e30d7f06fa/packages/react-instantsearch/src/widgets/RefinementList.tsx#L126C8-L126C28) is used, but [missing some of the options](https://github.com/algolia/instantsearch/blob/master/packages/react-instantsearch/src/ui/SearchBox.tsx#L73-L76).

**Comment by Haroenv** [comment]
This makes sense to me, you can open a pull request which adds `searchableSubmitIconComponent` etc and pass it on to searchbox.

Alternatively you can of course also take the code which is in RefinementList.tsx and edit it for your use case.

In the future it may make sense to have a full component option for the SearchBox as well, would need to debate that
